search/api.py

```diff
@@ -0,0 +1,261 @@
+    """
+    conn = get_conn(verify=False)
+
+    index_name = settings.ELASTICSEARCH_INDEX
+    if conn.indices.exists_type(index=index_name, doc_type=USER_DOC_TYPE):
+        conn.indices.delete_mapping(index=index_name, doc_type=USER_DOC_TYPE)
+
+    # A string which must be indexed exactly as is and not stemmed for full text search (the default)
+    not_analyzed_string_type = {
+        'type': 'string',
+        'index': 'not_analyzed',
+    }
+    bool_type = {'type': 'boolean'}
+    date_type = {'type': 'date', 'format': 'date'}
+    long_type = {'type': 'long'}
+
+    mapping = Mapping(USER_DOC_TYPE)
+    mapping.field("id", "long")
+    mapping.field("profile", "nested", properties={
+        'account_privacy': not_analyzed_string_type,
+        'agreed_to_terms_of_service': bool_type,
+        'birth_city': not_analyzed_string_type,
+        'birth_country': not_analyzed_string_type,
+        'birth_state_or_territory': not_analyzed_string_type,
+        'city': not_analyzed_string_type,
+        'country': not_analyzed_string_type,
+        'date_of_birth': date_type,
+        'education': {'type': 'nested', 'properties': {
+            'degree_name': not_analyzed_string_type,
+            'field_of_study': not_analyzed_string_type,
+            'graduation_date': date_type,
+            'id': long_type,
+            'online_degree': bool_type,
+            'school_city': not_analyzed_string_type,
+            'school_country': not_analyzed_string_type,
+            'school_name': not_analyzed_string_type,
+            'school_state_or_territory': not_analyzed_string_type,
+        }},
+        'email_optin': bool_type,
+        'filled_out': bool_type,
+        'first_name': not_analyzed_string_type,
+        'gender': not_analyzed_string_type,
+        'has_profile_image': bool_type,
+        'last_name': not_analyzed_string_type,
+        'preferred_language': not_analyzed_string_type,
+        'preferred_name': not_analyzed_string_type,
+        'pretty_printed_student_id': not_analyzed_string_type,
+        'profile_url_full': not_analyzed_string_type,
+        'profile_url_large': not_analyzed_string_type,
+        'profile_url_medium': not_analyzed_string_type,
+        'profile_url_small': not_analyzed_string_type,
+        'username': not_analyzed_string_type,
+        'work_history': {'type': 'nested', 'properties': {
+            'city': not_analyzed_string_type,
+            'company_name': not_analyzed_string_type,
+            'country': not_analyzed_string_type,
+            'id': long_type,
+            'industry': not_analyzed_string_type,
+            'position': not_analyzed_string_type,
+            'start_date': date_type,
+            'state_or_territory': not_analyzed_string_type,
+        }},
+    })
+    mapping.save(index_name)
+
+
+def create_mappings():
+    """
+    Create all mappings, deleting existing mappings if they exist.
+    """
+    create_user_mapping()
```

Comment:
do we have a reason for both `create_mappings` and `create_user_mapping` to exist?